### PR TITLE
Add llama.cpp GGUF backend with CUDA layer offload

### PR DESCRIPTION
**Pull request by iSparsh**
## Summary

Adds `--backend llamacpp` so `semif-score` can read last-position option logits from a local GGUF on Linux CUDA (or CPU). This is the CUDA follow-up to open #18, not a second CPU-only `llama-cpp-python` extra.

Torch stays BF16. MLX `--mlx-bits` stays Apple. These quantized scores are **not** the published 0.813 / 0.766 / 0.845 ladder. `results/phase1-summary.json` is unchanged.

## Why not #18's pip extra

`llama-cpp-python==0.3.35` CUDA wheels SIGILL in `ggml_cpu_init` on AVX2-only CPUs (i7-13700HX), and 0.3.35 ctypes `llama_model_params` is missing `lazy_mode` / `model_shared` versus current llama.cpp (Unsloth 0.4.1-dev). This PR binds `libllama` with ctypes and loads ggml CUDA/CPU backends from the same directory.

Happy to rebase CUDA `--llama-gpu-layers` onto #18 if you prefer one stack.

## Usage

```bash
semif-score --mode direct --backend llamacpp \
  --model Qwen/Qwen3.5-4B \
  --revision 851bf6e806efd8d0a36b00ddf55e13ccb7b8cd0a \
  --gguf /path/to/Qwen_Qwen3.5-4B-Q4_K_M.gguf \
  --llama-gpu-layers 99 \
  --input examples/decisions.jsonl \
  --output results-gguf-direct.jsonl
```

`--llama-lib` / `SEMIF_LLAMA_LIB` selects `libllama.so`. Default GPU layers: 99. `0` = CPU. Reranker stays torch.

## GPU evidence (bartowski Q4_K_M)

GGUF sha256 `13c16f426047e2de38cd075bdade4a7bcbc8c774384876f677740cda65f8a983` (3,013,027,808 bytes). Unsloth CUDA `libllama.so.0` 0.4.1-dev, `--llama-gpu-layers 99`, offloaded 34/34 layers.

Owned `examples/decisions.jsonl` via native `llama_get_logits_ith` (no generated tokens):

| id | choice | probabilities |
|---|---|---|
| support-1 | yes | 0.99962 / 0.00010 / 0.00027 |
| route-1 | account_access | 0.99996 / 0.00003 / 0.00001 |
| policy-1 | not_required | 0.06153 / 0.93459 / 0.00388 |

`support-1` answer token id 32 (`A`). Forward 0.14 s / 142 tok. Peak VRAM **3225 MiB**, process exit returns to **41 MiB**.

`pytest -q`: 42 passed, 2 skipped. Live GGUF test remains gated on `SEMIF_LLAMACPP_GGUF`.

### CODE REVIEW

**Comment by iSparsh** [comment]
My agent was misconfigured, apologies for this as I meant to work on my local clone
